public/app/core/utils/explore.ts

```diff
@@ -106,6 +106,7 @@ export async function getExploreUrl(
         ...state,
         datasource: panelDatasource.name,
         queries: exploreTargets.map(t => ({ ...t, datasource: panelDatasource.name })),
+        originPanel: panel.id,
```

Comment:
I never reach this code when using this with a Prometheus datasource. It's using the `if` branch above.